ddtrace/tracer/sampler.go

```diff
@@ -158,49 +158,115 @@ func (ps *prioritySampler) apply(spn *span) {
+		}
+	}
 	return &rulesSampler{
-		rules:      rules,
-		globalRate: globalSampleRate(),
-		limiter:    newRateLimiter(),
+		traceRulesSampler:      newTraceRulesSampler(traceRules),
+		singleSpanRulesSampler: newSingleSpanRulesSampler(spanRules),
 	}
 }
 
-// samplingRulesFromEnv parses sampling rules from the DD_TRACE_SAMPLING_RULES
-// environment variable.
+const (
+	// SamplingRuleTrace specifies that if a span matches a sampling rule of this type,
+	// an entire trace might be kept based on the given span and sent to the agent.
+	SamplingRuleTrace = iota
+	// SamplingRuleSingleSpan specifies that if a span matches a sampling rule,
+	// an individual span might be kept and sent to the agent.
+	SamplingRuleSingleSpan
+)
+
+// samplingRulesFromEnv parses sampling rules from the DD_TRACE_SAMPLING_RULES,
+// DD_SPAN_SAMPLING_RULES and DD_SPAN_SAMPLING_RULES_FILE environment variables.
 func samplingRulesFromEnv() ([]SamplingRule, error) {
+	var errs []string
+	var rules []SamplingRule
 	rulesFromEnv := os.Getenv("DD_TRACE_SAMPLING_RULES")
+	if rulesFromEnv != "" {
+		traceRules, err := processSamplingRules([]byte(rulesFromEnv), SamplingRuleTrace)
+		if err != nil {
+			errs = append(errs, err.Error())
+		}
+		rules = append(rules, traceRules...)
+	}
+	spanRules, err := spanSamplingRulesFromEnv()
+	if err != nil {
+		errs = append(errs, err.Error())
+	}
+	rules = append(rules, spanRules...)
+	if len(errs) != 0 {
+		return rules, fmt.Errorf("%s", strings.Join(errs, "\n\t"))
+	}
+	return rules, nil
+}
+
+// spanSamplingRulesFromEnv parses sampling rules from the DD_SPAN_SAMPLING_RULES and
+// DD_SPAN_SAMPLING_RULES_FILE environment variables.
+func spanSamplingRulesFromEnv() ([]SamplingRule, error) {
+	var errs []string
+	rulesFromEnv := os.Getenv("DD_SPAN_SAMPLING_RULES")
 	if rulesFromEnv == "" {
 		return nil, nil
 	}
-	jsonRules := []struct {
-		Service string      `json:"service"`
-		Name    string      `json:"name"`
-		Rate    json.Number `json:"sample_rate"`
-	}{}
-	err := json.Unmarshal([]byte(rulesFromEnv), &jsonRules)
+	rules, err := processSamplingRules([]byte(rulesFromEnv), SamplingRuleSingleSpan)
+	if err != nil {
+		errs = append(errs, err.Error())
+	}
+	rulesFile := os.Getenv("DD_SPAN_SAMPLING_RULES_FILE")
+	if len(rules) != 0 {
+		if rulesFile != "" {
+			log.Warn("DIAGNOSTICS Error(s): DD_SPAN_SAMPLING_RULES is available and will take precedence over DD_SPAN_SAMPLING_RULES_FILE")
+		}
+		return rules, err
+	}
+	if rulesFile != "" {
+		rulesFromEnvFile, err := os.ReadFile(rulesFile)
+		if err != nil {
+			log.Warn("Couldn't read file from DD_SPAN_SAMPLING_RULES_FILE")
+			return nil, err
+		}
+		rules, err = processSamplingRules(rulesFromEnvFile, SamplingRuleSingleSpan)
+		if err != nil {
+			errs = append(errs, err.Error())
+		}
+	}
+	if len(errs) != 0 {
+		return rules, fmt.Errorf("%s", strings.Join(errs, "\n\t"))
+	}
+	return rules, nil
+}
+
+func processSamplingRules(b []byte, spanType int) ([]SamplingRule, error) {
```

Comment:
```suggestion
// unmarhsalSamplingRules unmarhals JSON from b and returns the sampling rules found, attributing
// the type t to them. If any errors are occurred, they are returned.
func unmarshalSamplingRules(b []byte, t SamplingRuleType) ([]SamplingRule, error) {
```
Maybe you can phrase better than me 🙂